src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -54,6 +58,16 @@ public AddressBook toModelType() throws IllegalValueException {
             }
             addressBook.addEatery(eatery);
         }
+
+        addressBook.toggle();
+        for (JsonAdaptedEatery jsonAdaptedEatery : todos) {
+            Eatery eatery = jsonAdaptedEatery.toModelType();
+            if (addressBook.hasEatery(eatery)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_EATERY);
+            }
+            addressBook.addEatery(eatery);
+        }
+
         return addressBook;
```

Comment:
Does the address book mode need to be toggled back before returning?

Reply:
There is no need to, as in the MainApp class, a new AddressBook will be generated during the initialising of the modelManager. Therefore, the default mode when the application first start to run will be in main mode